nox/virtualenv.py

```diff
@@ -196,11 +296,22 @@ def create(self):
             logger.debug("Re-using existing virtualenv at {}.".format(self.location))
             return False
 
-        cmd = [sys.executable, "-m", "virtualenv", self.location]
+        cmd = [self._runtime_interpreter]
+        if not _FROZEN:
+            cmd.extend(["-m", "virtualenv"])
+        else:
+            # Nox was frozen with PyInstaller
+            # We have 2 options, import virtualenv and patch sys.argv to simlate
+            # a normal CLI call, or, find a python interpreter matching the one
+            # Nox was frozen with, or close enough, and feed it the path to the,
+            # also frozen, virtualenv.py module
+            cmd.append(os.path.join(sys._MEIPASS, "site-packages", "virtualenv.py"))
```

Comment:
Can you provide documentation explaining why this is the way it must be done? ISTM that `python -m` would be **perfect** for the "frozen" case so I'm unclear why it isn't supported. It's also a bit worrying that this relies on `_MEIPASS` which is non-public.

Reply:
https://pyinstaller.readthedocs.io/en/v3.3.1/runtime-information.html#using-file-and-sys-meipass

As you mention, `python -m` would be the perfect choice, if not for the fact that the `python` referred is `sys.executable` which, on a frozen single binary, it's the actual frozen binary.
That leaves us with a few options:
1. Have the frozen nox accept the `-m` argument. Then use that value to actually import `virtualenv`, patch `sys.argv` to fake a regular call to it(`python virtualenv.py`), then call it's `main()` function. This has many potential failure points.
2. Provide an upstream change to virtualenv to have it's `main()` function accept a list of args, `def main(args=None)`. If accepted, we would include virtualenv in the frozen nox binary and we would just import it and provide reasonable `args` for it to work, without munging with `sys.argv`. This is likely the best option, but it needs to be accepted upstream and doesn't solve the immediate problem.
3. Include `virtualenv.py` as a data file in the frozen binary, and call it with an available runtime python interpreter. This is the path followed here.